bokeh/core/property/bases.py

```diff
@@ -163,20 +164,21 @@ def readonly(self):
         return self._readonly
 
     def matches(self, new, old):
-        # XXX: originally this code warned about not being able to compare values, but that
-        # doesn't make sense, because most comparisons involving numpy arrays will fail with
-        # ValueError exception, thus warning about inevitable.
-        try:
-            if new is None or old is None:
-                return new is old           # XXX: silence FutureWarning from NumPy
-            else:
-                return new == old
-        except (KeyboardInterrupt, SystemExit):
-            raise
-        except Exception:
-            # if we cannot compare (e.g. arrays) just punt return False for match
-            pass
-        return False
+        ''' Whether two parameters match values.
+
+        If either ``new`` or ``old`` is a NumPy array, then the result of
+        ``np.array_equal`` will determine if the values match..
+
+        Otherwise, the result of standard Python equality will be returned.
+
+        Returns:
+            True, if new and old match, False otherwise
+
+        '''
+        if isinstance(new, np.ndarray) or isinstance(old, np.ndarray):
+            return np.array_equal(new, old)
+
+        return new == old
```

Comment:
`array_equal` avoids the `FutureWarning` (except erroneously in one probably uncommon case: https://github.com/numpy/numpy/issues/9210) so this new code seems both more robust as well as simpler. Also by using `array_equal` I don't think there are any "expected" exceptions, so we can remove the `try...except` including the bad `except Exception` block

It's worth noting that the old notes about never comparing arrays were wrong, the `old==new` *did* compare arrays but returned bool arrays. We have existing tests that depend on arrays getting compared. All of them pass unchanged with this new code.